pkg/builder/deployment_builder.go

```diff
@@ -23,6 +23,39 @@ const (
 	maxScaleRuntimeConfigMountPath = "/var/lib/maxscale/maxscale.cnf.d"
 )
 
+func (b *Builder) exporterPodTemplate(mariadb *mariadbv1alpha1.MariaDB, selectorLabels map[string]string) (*corev1.PodTemplateSpec, error) {
+	container, err := buildExporterContainer(mariadb)
+	if err != nil {
+		return nil, fmt.Errorf("error building exporter container: %v", err)
+	}
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: selectorLabels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				container,
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: deployConfigVolume,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: mariadb.MetricsConfigSecretKeyRef().Name,
+						},
+					},
+				},
+			},
+			SecurityContext:           mariadb.Spec.Metrics.Exporter.PodSecurityContext,
+			Affinity:                  mariadb.Spec.Metrics.Exporter.Affinity,
+			NodeSelector:              mariadb.Spec.Metrics.Exporter.NodeSelector,
+			Tolerations:               mariadb.Spec.Metrics.Exporter.Tolerations,
+			PriorityClassName:         priorityClass(mariadb.Spec.Metrics.Exporter.PriorityClassName),
+			TopologySpreadConstraints: mariadb.Spec.Metrics.Exporter.TopologySpreadConstraints,
```

Comment:
I added the fields that I thought would be useful here. I'm only setting `securityContext` and `nodeSelector` myself.

Reply:
Fantastic! Sounds reasonable to me as well.